src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

```diff
@@ -58,6 +58,7 @@ public function process(ContainerBuilder $container): void
             $defaultContext = $container->getParameter('serializer.default_context');
             $this->bindDefaultContext($container, array_merge($normalizers, $encoders), $defaultContext);
             $container->getParameterBag()->remove('serializer.default_context');
+            $container->getDefinition('serializer')->replaceArgument(2, $defaultContext);
```

Comment:
instead of a positional argument, let's use a named one:
```suggestion
            $container->getDefinition('serializer')->setArgument('$defaultContext', $defaultContext);
```

this will make the container fail at compile-time in case the `serrializer` service is overriden. Better than a runtime error.